server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java

```diff
@@ -324,24 +320,17 @@ public void test_handshake_precedes_stream_version_for_compatibility() throws Ex
      */
     @Test
     public void test_handshake_checks_minimum_compatible_version_if_normal_version_is_not_compatible() throws Exception {
-        Version localVersion = Version.V_5_10_1;
-        NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry(Collections.emptyList());
-        TransportHandshaker handshaker = new TransportHandshaker(localVersion, threadPool, (n, c, r, v) -> {});
-        TransportKeepAlive keepAlive = new TransportKeepAlive(threadPool, (c, b) -> channel.writeAndFlush(Unpooled.wrappedBuffer(b)));
-        OutboundHandler outboundHandler = new OutboundHandler("node", localVersion, new StatsTracker(), threadPool,
-            BigArrays.NON_RECYCLING_INSTANCE);
-        InboundHandler handler = new InboundHandler(threadPool, outboundHandler, namedWriteableRegistry, handshaker, keepAlive, requestHandlers,
-            responseHandlers);
-
-
+        Version localVersion = Version.V_6_0_0;
+        Version remoteVersion = Version.V_5_10_1;
+        InboundHandler handler = handlerForVersion(remoteVersion);
```

Comment:
Spot while backporting, we essntially swapped 
`(local, remote)` from `(V_5_10_1, CURRENT = 6.0.0)` to `(6.0.0, 5.10.1)`

i think idea is that order doesn't matter for handshake but on 5.10 we run into 4.3 version check ([comment](https://github.com/crate/crate/pull/17439/files#r1954491420))

Reply:
Just the variable naming was corrected, the behavior should be the same. I'll check the backport.